### PR TITLE
Dynamic cusum filter, z_score filter

### PR DESCRIPTION
**Pull request by proskurin**
# Description

Improve cusum filter so that it can take not only fixed threshold but also dynamic thresholds as as series, add Z-score filter

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- test_filters.py

**Test Configuration**:
* Ubuntu 16.04
* PyCharm

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: mlfinlab/filters/filters.py [hunk 1]
```diff
@@ -44,26 +38,57 @@ def cusum_filter(raw_time_series, threshold, time_stamps=True):
...
 
     # Return DatetimeIndex or list
     if time_stamps:
         event_timestamps = pd.DatetimeIndex(t_events)
         return event_timestamps
 
     return t_events
+
+
+def z_score_filter(raw_time_series, mean_window, std_window, z_score=3, time_stamps=True):
```
Comment (Jackal08) [inline mlfinlab/filters/filters.py]: Very cool!

File: mlfinlab/tests/test_filters.py [hunk 2]
```diff
@@ -51,3 +51,31 @@ def test_cusum_filter(self):
                     cusum = max(np.abs(sneg), spos)
 
                     self.assertTrue(cusum >= threshold)
+
+    def test_dynamic_cusum_filter(self):
```
Comment (Jackal08) [inline mlfinlab/tests/test_filters.py]: Are there any extra edge cases you want us to test?

File: mlfinlab/tests/test_filters.py [hunk 3]
```diff
@@ -37,7 +37,7 @@ def test_cusum_filter(self):
                 cusum_events = cusum_filter(self.data['close'], threshold=threshold, time_stamps=timestamp)
```
Comment (Jackal08) [inline mlfinlab/tests/test_filters.py]: Is this new method able to behave in the same way as the old one? 

I really like the idea here, its one I think we all wanted implemented when we started this project. Glad to see the change. I bring up the reference to the old funciton because the CUSUM filter is linked to Ch17 on structural breaks.

File: mlfinlab/filters/filters.py (hunk 1 above)
Comment (Jackal08) [inline mlfinlab/filters/filters.py]: Please remember to add this function to the read-the-docs.

File: mlfinlab/tests/test_filters.py (hunk 3 above)
Comment (proskurin) [reply mlfinlab/tests/test_filters.py]: Yeah, if fixed threshold is passed, behavior is the same. I didn't change your tests and passed them

File: mlfinlab/tests/test_filters.py (hunk 2 above)
Comment (proskurin) [reply mlfinlab/tests/test_filters.py]: The first part of test (I mean yours) tests edge cases, I've decided that for dynamic part simply testing the resulting shape would be enough
